libfwupdplugin/fu-context.c

```diff
@@ -1495,26 +1495,17 @@ fu_context_get_esp_volumes(FuContext *self, GError **error)
 
 	/* nothing found */
 	if (priv->esp_volumes->len == 0) {
-		g_autofree gchar *base = NULL;
-		g_autofree gchar *path = NULL;
 		g_autoptr(GPtrArray) devices = NULL;
 
-		/* udisks2 is working */
-		devices = fu_common_get_block_devices(NULL);
+		/* check if udisks2 is working */
+		devices = fu_common_get_block_devices(error);
 		if (devices != NULL) {
```

Comment:
maybe flip this around? i.e. `if (devices == NULL); return NULL` then fallback to the `No ESP or BDP found`